eng/Versions.props

```diff
@@ -144,6 +140,10 @@
     <SystemThreadingTasksExtensionsVersion>4.5.4</SystemThreadingTasksExtensionsVersion>
     <SystemValueTupleVersion>4.5.0</SystemValueTupleVersion>
     <runtimenativeSystemIOPortsVersion>9.0.0-preview.2.24111.9</runtimenativeSystemIOPortsVersion>
+    <!-- Keep toolset versions in sync with dotnet/msbuild and dotnet/sdk -->
+    <SystemReflectionMetadataToolsetVersion>8.0.0</SystemReflectionMetadataToolsetVersion>
+    <SystemReflectionMetadataLoadContextToolsetVersion>8.0.0</SystemReflectionMetadataLoadContextToolsetVersion>
+    <SystemCollectionsImmutableToolsetVersion>8.0.0</SystemCollectionsImmutableToolsetVersion>
```

Comment:
I think source-build needs to be able to override these versions. For that, we need Version.Details.xml entries and Versions.props updates, i.e.:

```
<SystemReflectionMetadataToolsetVersion>8.0.0</SystemReflectionMetadataToolsetVersion>
<SystemReflectionMetadataToolsetVersion Condition="'$(DotNetBuildFromSource)' == 'true'">$(SystemReflectionMetadataVersion)</SystemReflectionMetadataToolsetVersion>
```

@dotnet/source-build-internal please double check my suggestion.

Reply:
This isn't necessary for SCI and SRM which shouldn't be referenced on .NETCoreApp. But it is necessary for S.R.MetadataLoadContext. Revert the Version.details.xml changes and the Versions.props change for it.

Reply:
ah ok that makes sense. I removed them because darc was updating the versions but your suggestion makes sense.

Reply:
AFAIK these tasks are run in the full product source-build so we must not use SBRPs. Anyway SCI and SRM are inbox on .NETCoreApp and hence we don't need to special case them for source-build.

Reply:
We never have S.R.MetadataLoadContext.8.0.0 in SBRP, because of the Versions.Details.xml entry, this was being lifted to the current version in the product SB.